flink-runtime/src/main/java/org/apache/flink/runtime/dispatcher/Dispatcher.java

```diff
@@ -362,6 +362,7 @@ private boolean isPartialResourceConfigured(JobGraph jobGraph) {
 		jobManagerRunnerFutures.put(jobGraph.getJobID(), jobManagerRunnerFuture);
 
 		return jobManagerRunnerFuture
+			.thenApply(FunctionUtils.uncheckedFunction(this::startJobManagerRunner))
```

Comment:
This change causes a slight semantic change. Before `jobManagerRunnerFutures` contained only a completed `JobManagerRunner` if the `JobManagerRunner` was actually running. This is now no longer true. Hence it could happen that the `Dispatcher` receives a request for a job which then being forwarded to the not yet started `JobMaster`.

In order to solve this problem and the one you are solving here, we could move the registration of the `JobManagerRunner#getResultFuture` to here but leave the starting in `createJobManagerRunner`. Wdyt?

Reply:
> it could happen that the Dispatcher receives a request for a job which then being forwarded to the not yet started `JobMaster`.

We always forward request to `JobMaster` leader by `jobManagerRunnerFuture.thenCompose(JobManagerRunner::getLeaderGatewayFuture)` and thus I think we don't have a potential risk to execute a request on a un-started `JobMaster`?

Reply:
I think you are right. Then this fix is good to go :-)